sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_transport.py

```diff
@@ -551,7 +552,11 @@ def _wrap_socket_sni(  # pylint: disable=no-self-use
         }
 
         # TODO: We need to refactor this.
-        sock = ssl.wrap_socket(**opts)  # pylint: disable=deprecated-method
+        try:
+            sock = ssl.wrap_socket(**opts)  # pylint: disable=deprecated-method
+        except FileNotFoundError as exc:
+            exc.filename = {"ca_certs": ca_certs}
```

Comment:
adding this line so we can raise FileNotFoundError for the connection_verify cert file, specifically. Don't want to necessarily surface this type of error for other files that may also raise a FileNotFoundError.

Reply:
But it doesn't look like we're check that this _was_ raised because of the connection_verify cert file?

Reply:
so this FileNotFoundError doesn't have the name of the file that was not found stored in any of the attributes in the error, except the error message. So, unless I parse the error message, I'm not able to check against the cert file.

However, from the client side, this is the only file that the user can specify/pass through the kwargs. So, I assumed that the cert file would be the issue here. Maybe a bad assumption though?

Reply:
It's probably a reasonable assumption... Do we allow kwargs passthrough to the transport? I don't think we do, so I don't think we'd ever have a situation was that `ssl_opts` could include some filepath we don't know about...

Reply:
No, we don't pass kwargs through.

connection_verify path is [passed into the SendClient](https://github.com/Azure/azure-sdk-for-python/blob/5013c2d8464a518ff4f2a6ca691f7913e1620c6e/sdk/eventhub/azure-eventhub/azure/eventhub/_transport/_pyamqp_transport.py#L253), and we set it to an attribute on the client. We then pass that through to the transport constructor [[here](https://github.com/Azure/azure-sdk-for-python/blob/5013c2d8464a518ff4f2a6ca691f7913e1620c6e/sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/client.py#L291)].

Reply:
Okay - though once again could we make sure things like this are _thoroughly_ documented with inline comments?

Reply:
yes, adding this now!